py-scripts/lf_webpage.py

```diff
@@ -1522,68 +1885,85 @@ def main():
-            list5G_bytes.extend(rx_bytes_val)
-            list5G_speed.extend(rx_rate_val)
-            list5G_urltimes.extend(url_times)
-            logger.info("%s %s %s %s", list5G, list5G_bytes, list5G_speed, list5G_urltimes)
-            final_dict['5G']['dl_time'] = list5G
-            min5.append(min(list5G))
-            final_dict['5G']['min'] = min5
-            max5.append(max(list5G))
-            final_dict['5G']['max'] = max5
-            avg5.append((sum(list5G) / args.num_stations))
-            final_dict['5G']['avg'] = avg5
-            final_dict['5G']['bytes_rd'] = list5G_bytes
-            final_dict['5G']['speed'] = list5G_speed
-            final_dict['5G']['url_times'] = list5G_urltimes
-        elif bands == "6G":
-            list6G.extend(uc_avg_val)
-            list6G_bytes.extend(rx_bytes_val)
-            list6G_speed.extend(rx_rate_val)
-            list6G_urltimes.extend(url_times)
-            final_dict['6G']['dl_time'] = list6G
-            min6.append(min(list6G))
-            final_dict['6G']['min'] = min6
-            max6.append(max(list6G))
-            final_dict['6G']['max'] = max6
-            avg6.append((sum(list6G) / args.num_stations))
-            final_dict['6G']['avg'] = avg6
-            final_dict['6G']['bytes_rd'] = list6G_bytes
-            final_dict['6G']['speed'] = list6G_speed
-            final_dict['6G']['url_times'] = list6G_urltimes
-        elif bands == "2.4G":
-            list2G.extend(uc_avg_val)
-            list2G_bytes.extend(rx_bytes_val)
-            list2G_speed.extend(rx_rate_val)
-            list2G_urltimes.extend(url_times)
-            logger.info("%s %s %s", list2G, list2G_bytes, list2G_speed)
-            final_dict['2.4G']['dl_time'] = list2G
-            min2.append(min(list2G))
-            final_dict['2.4G']['min'] = min2
-            max2.append(max(list2G))
-            final_dict['2.4G']['max'] = max2
-            avg2.append((sum(list2G) / args.num_stations))
-            final_dict['2.4G']['avg'] = avg2
-            final_dict['2.4G']['bytes_rd'] = list2G_bytes
-            final_dict['2.4G']['speed'] = list2G_speed
-            final_dict['2.4G']['url_times'] = list2G_urltimes
-        elif bands == "Both":
-            Both.extend(uc_avg_val)
-            Both_bytes.extend(rx_bytes_val)
-            Both_speed.extend(rx_rate_val)
-            Both_urltimes.extend(url_times)
-            final_dict['Both']['dl_time'] = Both
-            min_both.append(min(Both))
-            final_dict['Both']['min'] = min_both
-            max_both.append(max(Both))
-            final_dict['Both']['max'] = max_both
-            avg_both.append((sum(Both) / args.num_stations))
-            final_dict['Both']['avg'] = avg_both
-            final_dict['Both']['bytes_rd'] = Both_bytes
-            final_dict['Both']['speed'] = Both_speed
-            final_dict['Both']['url_times'] = Both_urltimes
+        if args.client_type == 'Real':  # for real clients
```

Comment:
This is pretty verbose but fine for now. I see a lot of what appears to be duplicated logic. Possibly this could be refactored into a helper function per band. No changes requested.